icu4j/main/core/src/main/java/com/ibm/icu/util/ULocale.java

```diff
@@ -675,8 +699,9 @@ public static ULocale getDefault(Category category) {
                 // time.
 
                 Locale currentDefault = Locale.getDefault();
-                if (!defaultLocale.equals(currentDefault)) {
-                    defaultLocale = currentDefault;
+                ULocale currentDefaultULocale = defaultULocale;
+                Locale defaultLocale = currentDefaultULocale == null ? null : currentDefaultULocale.locale;
+                if (!Objects.equals(defaultLocale, currentDefault)) {
```

Comment:
Here, too, the defaultULocale should only be null during class initialization. Except that this function has checked for that already right at the start, which means we cannot have defaultULocale == null now.

```suggestion
                if (!currentDefaultULocale.locale.equals(currentDefault)) {
```